ambry-account/src/main/java/com/github/ambry/account/mysql/DatasetDao.java

```diff
@@ -918,6 +959,47 @@ private Page<String> executeListAllValidDatasetsStatement(PreparedStatement stat
     }
   }
 
+  /**
+   * Execute listValidDatasetVersionsSql statement.
+   * @param statement the listValidDatasetVersionsSql statement.
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param datasetName the name of the dataset.
+   * @param pageToken the page token.
+   * @param versionSchema the dataset version schema.
+   * @return the page of the dataset versions.
+   * @throws SQLException
+   */
+  private Page<String> executeListAllValidDatasetVersionsStatement(PreparedStatement statement, int accountId,
+      int containerId, String datasetName, String pageToken, Dataset.VersionSchema versionSchema) throws SQLException {
+    ResultSet resultSet = null;
+    try {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, datasetName);
+      statement.setInt(4, DatasetVersionState.READY.ordinal());
+      statement.setString(5, pageToken);
+      statement.setString(6, pageToken);
```

Comment:
we should convert pageToken back to a version number before passing it to the query statement, we should do it before calling this execute method.

Reply:
Fixed.